lib/hanami/action/mime.rb

```diff
@@ -565,33 +560,41 @@ def content_type_with_charset
       # @see https://github.com/rack/rack/pull/659
       # @see https://github.com/hanami/controller/issues/59
       # @see https://github.com/hanami/controller/issues/104
+      # @see https://github.com/hanami/controller/issues/275
       def best_q_match(q_value_header, available_mimes)
-        values = ::Rack::Utils.q_values(q_value_header)
-        values = values.map do |req_mime, quality|
-          if req_mime == DEFAULT_ACCEPT
-            # See https://github.com/hanami/controller/issues/167
-            match = default_content_type
-          else
-            match = available_mimes.find { |am| ::Rack::Mime.match?(am, req_mime) }
-          end
+        values = []
+        ::Rack::Utils.q_values(q_value_header).each_with_index do |(req_mime, quality), index|
```

Comment:
Yes `values` is useless, I've let it on another line for readability but hesitated, so if you find it understandable enough, just chain to the end :+1: